specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/1.2-preview/iotcentral.json

```diff
@@ -0,0 +1,5643 @@
+            }
+          },
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "#/definitions/Error"
+            }
+          }
+        }
+      },
+      "put": {
+        "operationId": "Devices_CreateRelationship",
+        "summary": "Given the ID for a device and a relationship ID associated with this device, create a new relationship for between the given device and a second device specified in the body.",
+        "description": "Create a device relationship",
+        "x-ms-examples": {
+          "Create a new device relationship": {
+            "$ref": "./examples/devices_relationships_create.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/Version"
+          },
+          {
+            "$ref": "#/parameters/DeviceId"
+          },
+          {
+            "$ref": "#/parameters/RelationshipId"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "description": "Device relationship body.",
+            "schema": {
+              "$ref": "#/definitions/DeviceRelationship"
+            },
+            "required": true
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/DeviceRelationship"
+            }
+          },
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "#/definitions/Error"
+            }
+          }
+        }
+      },
+      "patch": {
+        "operationId": "Devices_UpdateRelationship",
+        "summary": "Patch a given relationship given the relationship ID and a given device ID.",
+        "description": "Update device relationship",
+        "x-ms-examples": {
+          "Update a device relationship": {
+            "$ref": "./examples/devices_relationships_update.json"
+          }
+        },
+        "consumes": [
+          "application/json",
+          "application/json-patch+json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/Version"
+          },
+          {
+            "$ref": "#/parameters/DeviceId"
+          },
+          {
+            "$ref": "#/parameters/RelationshipId"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "description": "Device relationship patch body.",
+            "schema": {
+              "type": "object"
+            },
+            "required": true
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/DeviceRelationship"
+            }
+          },
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "#/definitions/Error"
+            }
```

Comment:
Error response should contain a x-ms-error-code header.

Reply:
Right now we don't have implementation of x-ms-error-code header, will skip now.